public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx

```diff
@@ -207,50 +232,71 @@ export class ShareSnapshot extends PureComponent<Props, State> {
+    });
+
+    const timeoutDescriptionTranslation = t({
+      id: 'share-modal.snapshot.timeout-description',
+      message: `You might need to configure the timeout value if it takes a long time to collect your dashboard
+      metrics.`,
+    });
+
     return (
       <>
         <div>
           <p className="share-modal-info-text">
-            A snapshot is an instant way to share an interactive dashboard publicly. When created, we strip sensitive
-            data like queries (metric, template, and annotation) and panel links, leaving only the visible metric data
-            and series names embedded in your dashboard.
+            <Trans id="share-modal.snapshot.info-text-1">
+              A snapshot is an instant way to share an interactive dashboard publicly. When created, we strip sensitive
+              data like queries (metric, template, and annotation) and panel links, leaving only the visible metric data
+              and series names embedded in your dashboard.
+            </Trans>
           </p>
           <p className="share-modal-info-text">
-            Keep in mind, your snapshot <em>can be viewed by anyone</em> that has the link and can access the URL. Share
-            wisely.
+            <Trans id="share-modal.snapshot.info-text-2">
+              Keep in mind, your snapshot <em>can be viewed by anyone</em> that has the link and can access the URL.
+              Share wisely.
+            </Trans>
           </p>
         </div>
-        <Field label="Snapshot name">
+        <Field label={snapshotNameTranslation}>
           <Input id="snapshot-name-input" width={30} value={snapshotName} onChange={this.onSnapshotNameChange} />
         </Field>
-        <Field label="Expire">
+        <Field label={expireTranslation}>
           <Select
             inputId="expire-select-input"
             width={30}
-            options={expireOptions}
+            options={this.expireOptions}
             value={selectedExpireOption}
             onChange={this.onExpireChange}
           />
         </Field>
-        <Field
-          label="Timeout (seconds)"
-          description="You might need to configure the timeout value if it takes a long time to collect your dashboard
-            metrics."
-        >
+        <Field label={timeoutTranslation} description={timeoutDescriptionTranslation}>
           <Input id="timeout-input" type="number" width={21} value={timeoutSeconds} onChange={this.onTimeoutChange} />
         </Field>
 
         <Modal.ButtonRow>
           <Button variant="secondary" onClick={onDismiss} fill="outline">
-            Cancel
+            <Trans id="share-modal.snapshot.copy-button">Cancel</Trans>
           </Button>
           {externalEnabled && (
             <Button variant="secondary" disabled={isLoading} onClick={this.createSnapshot(true)}>
-              {sharingButtonText}
+              <Trans id="share-modal.snapshot.sharing-button">{sharingButtonText}</Trans>
```

Comment:
The translators can't translate this 😅  just remove the `Trans` component, do we know if we have to translate this from somewhere else?